noir-projects/aztec-nr/aztec/src/utils/conversion/fields_to_bytes.nr

```diff
@@ -0,0 +1,87 @@
+/// These functions are used to facilitate the conversion of log plaintext represented as fields into bytes and back.
+///
+/// `fields_to_bytes` uses fixed-size arrays since encryption contexts have compile-time size information.
+/// `fields_from_bytes` uses BoundedVec for flexibility in unconstrained contexts where sizes are dynamic.
+///
+/// Together they provide bidirectional conversion between fields and bytes.
+
+/// Converts an input array of fields into a single array of bytes.
+/// Each field is converted to a 32-byte big-endian array.
+/// Since a field is ~254 bits, you'll end up with a subtle 2-bit "gap" at the big end,
+/// every 32 bytes. Be careful that such a gap doesn't leak information!
+pub fn fields_to_bytes<let N: u32>(fields: [Field; N]) -> [u8; 32 * N] {
+    let mut bytes = [0; 32 * N];
+
+    for i in 0..N {
+        let field = fields[i];
+        let field_as_bytes: [u8; 32] = field.to_be_bytes();
+
+        for j in 0..32 {
+            bytes[i * 32 + j] = field_as_bytes[j];
+        }
+    }
+
+    bytes
+}
+
+/// Converts an input BoundedVec of bytes into a BoundedVec of fields.
+///
+/// The input bytes are chunked into chunks of 32 bytes. Each 32-byte chunk is viewed as big-endian, and is converted
+/// into a Field.
+/// For example, [1, 10, 3, ..., 0] (32 bytes) is encoded as [1 * 256^31 + 10 * 256^30 + 3 * 256^29 + ... + 0]
+/// Note: N must be a multiple of 32 bytes
+pub fn fields_from_bytes<let N: u32>(bytes: BoundedVec<u8, N>) -> BoundedVec<Field, N / 32> {
+    // Assert that input length is a multiple of 32
+    assert(bytes.len() % 32 == 0, "Input length must be a multiple of 32");
+
+    let mut fields = BoundedVec::new();
+
+    // Since input length is a multiple of 32, we can simply process all chunks fully
+    for i in 0..bytes.len() / 32 {
+        let mut field = 0;
+        for j in 0..32 {
+            // Shift the existing value left by 8 bits and add the new byte
+            field = field * 256 + bytes.get(i * 32 + j) as Field;
+        }
```

Comment:
Are we not missing an overflow check here? We're not checking that when adding the last byte into the field we're not going over the maximum value.

Reply:
Good catch. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/12581/commits/98a5ab0e9eaad29eb9132139577ddd33ae917cb7